src/com/akshay/captain/jira.main.kts

```diff
@@ -0,0 +1,240 @@
+@file:Repository("https://repo.maven.apache.org/maven2")
+@file:Repository("https://packages.atlassian.com/maven/repository/public")
+@file:DependsOn("com.slack.api:slack-api-client:1.11.0")
+@file:DependsOn("net.rcarz:jira-client:0.5")
+@file:DependsOn("com.squareup.okhttp3:okhttp:4.9.3")
+
+import Jira_main.Platform
+import Jira_main.TicketType
+import net.rcarz.jiraclient.BasicCredentials
+import net.rcarz.jiraclient.Issue
+import net.rcarz.jiraclient.IssueType
+import net.rcarz.jiraclient.JiraClient
+import net.sf.json.JSONObject
+import okhttp3.*
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
+
+
+//region Configuration
+private val changeEmoji = "✨"
+private val techChangesEmoji = "🤖"
+private val bugEmoji = "🐛"
+private val instrumentationEmoji = "📈"
+//endregion
+
+//region constant
+private val changes = "Changes"
+private val techChanges = "Tech changes"
+private val bugFixes = "Bug fixes"
+private val instrumentation = "Instrumentation"
+
+//Jira Variables
+private val clueJiraUrl = "https://biowink.atlassian.net"
+private val readyForReleaseFilter = "filter=15768"
+private val firstIndex = 0
+private val authorization = "Authorization"
+private val jsonType = "application/json; charset=utf-8"
+private val versionEndPoint = "/rest/api/2/version"
+
+//Platform
+private val android = "android"
+private val ios = "ios"
+main()
+
+fun main() {
+    // 1. Take platform from path or by default it takes android
+    val platform = args.getOrNull(firstIndex)?.toPlatform() ?: Platform.Android
+
+    // createReleaseVersionInJira()
+    val cred = BasicCredentials("userEmail", "userToken")
+    val jira = JiraClient(clueJiraUrl, cred)
+    val releaseNotes = mutableListOf<String>()
+
+    try {
+        val result: Issue.SearchResult = jira.searchIssues(readyForReleaseFilter)
+
+        val tickets = result.generateTickets(platform)
+
+        // 2. Print all changes
+        tickets
+            .filter { it.ticketType.isChange() }
+            .ifNotEmpty { releaseNotes.add(changes) }
+            .map { releaseNotes.add("$changeEmoji ${it.key} \t ${it.summary}") }
+
+        // 3. Print all tech changes
+        tickets
+            .filter { it.ticketType is TicketType.Chapter }
+            .ifNotEmpty { releaseNotes.add(techChanges) }
+            .map { releaseNotes.add("$techChangesEmoji ${it.key} \t ${it.summary}") }
+
+        // 4. Print all bug fixes
+        tickets
+            .filter { it.ticketType is TicketType.Bug }
+            .ifNotEmpty { releaseNotes.add(bugFixes) }
+            .map { releaseNotes.add("$bugEmoji ${it.key} \t ${it.summary}") }
+
+        // 5. Print all instrumentation
+        tickets
+            .filter { it.ticketType is TicketType.Analytics }
+            .ifNotEmpty { releaseNotes.add(instrumentation) }
+            .map { releaseNotes.add("$instrumentationEmoji ${it.key} \t ${it.summary}") }
+
```

Comment:
I would abstract this to a function that gets a type to not repeat almost the same code 4 times

Reply:
This was same as other code, hence never touched it